heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -39,12 +40,105 @@
 import com.twitter.heron.spi.utils.ShellUtils;
 
 public final class SchedulerUtils {
-  public static final int PORTS_REQUIRED_FOR_EXECUTOR = 9;
+  public static final int PORTS_REQUIRED_FOR_EXECUTOR = ExecutorPorts.getRequiredNumOfPorts();
   public static final int PORTS_REQUIRED_FOR_SCHEDULER = 1;
   public static final String SCHEDULER_COMMAND_LINE_PROPERTIES_OVERRIDE_OPTION = "P";
 
   private static final Logger LOG = Logger.getLogger(SchedulerUtils.class.getName());
 
+  /**
+   * Enum that defines the type of ports that an heron executor needs
+   */
+  public static class ExecutorPorts extends HashMap<ExecutorPorts.ExecutorPortNames, Object> {
+    private static final long serialVersionUID = 3594142561871208879L;
+
+    public enum ExecutorPortNames {
+      MASTER_PORT("master"),
+      TMASTER_CONTROLLER_PORT("tmaster-ctl"),
+      TMASTER_STATS_PORT("tmaster-stats"),
+      SHELL_PORT("shell-port"),
+      METRICS_MANAGER_PORT("metrics-mgr"),
+      SCHEDULER_PORT("scheduler"),
+      METRICS_CACHE_MASTER_PORT("metrics-cache-m"),
+      METRICS_CACHE_STATS_PORT("metrics-cache-s"),
+      CHECKPOINT_MANAGER_PORT("ckptmgr");
+
+      private final String name;
+
+      ExecutorPortNames(String name) {
+        this.name = name;
+      }
+
+      public String toString() {
```

Comment:
Instead of overloading `toString`, I think it would be better to add a `getter` method

Reply:
I think its more nature to override toString.  If I was to print ExecutorPortNames, I can just do 

System.out.println(ExecutorPortNames.MASTER_PORT) instead of something like System.out.println(ExecutorPortNames.MASTER_PORT.getName())

Reply:
I will not come to know that `toString` is returning the id of the port without looking at the code. One might also assume `System.out.println(ExecutorPortNames.MASTER_PORT)` prints port name and value too.

Reply:
Please do not use the value of `toString()` for any business logic. The `toString` value should only ever be used to log messages **and nothing else**. Overriding `toString()` for logic is an anti-pattern. Doing so violates encapsulation and provides and unclear API to consumers. Also future developers could find that the logging of the object is unclear (e.g., logging just an int for a Port enum might not be desirable) and change it. This would break functionality. I have been bit by this and have spent hours debugging such design flaws. Please don't do this.